Libraries/libcar/Sources/Rendition.cpp

```diff
@@ -564,15 +527,23 @@ write() const
     strncpy(header.metadata.name, _fileName.c_str(), 128);
 
     // Create info segments
-
-    int nslices = SliceCountForLayout(_layout);
+    size_t nslices = _slices.size() > 0 ? _slices.size() : 1;
     size_t info_slices_size = sizeof(car_rendition_info_slices) + sizeof(struct car_rendition_info_slice) * nslices;
     struct car_rendition_info_slices *info_slices = (struct car_rendition_info_slices *)malloc(info_slices_size);
     info_slices->header.magic = car_rendition_info_magic_slices;
     info_slices->header.length = info_slices_size - sizeof(car_rendition_info_header);
-    // FIXME write slices
-    info_slices->nslices = nslices;
-    if (nslices == 1) {
+    if (_slices.size() > 0) {
+        int index = 0;
+        info_slices->nslices = _slices.size();
+        for (auto const &slice : _slices) {
+            info_slices->slices[index].x = slice.x;
+            info_slices->slices[index].y = slice.y;
+            info_slices->slices[index].width = slice.width;
+            info_slices->slices[index].height = slice.height;
+            index += 1;
```

Comment:
nit: maybe use `for (size_t i = 0; i < _slices.size(); i++)` so both sides use the same indexing?